proton-c/bindings/python/proton/__init__.py

```diff
@@ -2581,6 +2581,9 @@ def close(self):
     """
     self._update_cond()
     pn_connection_close(self._impl)
+    if hasattr(self, '_session_policy'):
```

Comment:
Out of curiosity, why isn't _session_policy a stable attribute of the object?  That is, could we simply assign None here?  I would prefer that a bit if it makes no difference.  This likely predates your change, but I usually try to avoid having to test for the presence of attributes in Python objects.